applications/ContactStructuralMechanicsApplication/tests/test_mortar_mapper.py

```diff
@@ -93,38 +93,62 @@ def __base_test_mapping(self, input_filename, num_nodes):
         contact_search.InitializeMortarConditions()
         contact_search.UpdateMortarConditions()
 
-        for node in self.model_part_slave.Nodes:
+        for node in self.model_part_master.Nodes:
             x = node.X 
             y = node.Y 
             z = node.Z
-            node.SetSolutionStepValue(KratosMultiphysics.PRESSURE, z)
-            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_X, x)
-            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_Y, y)
-            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_Z, z)
+            node.SetSolutionStepValue(KratosMultiphysics.TEMPERATURE, z)
+            node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X, x)
+            node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y, y)
+            node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z, z)
         del(node)
 
-        if (num_nodes == 3): 
-            self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.PRESSURE)
-            self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D3NVectorHistorical(self.main_model_part, KratosMultiphysics.VELOCITY)
+        map_parameters = KratosMultiphysics.Parameters("""
+        {
+            "echo_level"                       : 0,
+            "absolute_convergence_tolerance"   : 1.0e-9,
+            "relative_convergence_tolerance"   : 1.0e-4,
+            "max_number_iterations"            : 10,
+            "integration_order"                : 2
+        }
+        """)
+
+        if (pure_implicit == True):
+            #linear_solver = ExternalSolversApplication.SuperLUSolver()
+            linear_solver = KratosMultiphysics.SkylineLUFactorizationSolver()
+
+            if (num_nodes == 3): 
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters, linear_solver)
```

Comment:
accepting for now, however can u discuss with @philbucher  to use a common interface? (valid both for scalars and vectors)

Reply:
I will discuss with  him on Tuesday, but this is more like an example and something that @annasophie91 needs ASAP for her master thesis, so I think we can modify the interface in a future